packages/pwa-kit-dev/src/configs/webpack/config.js

```diff
@@ -338,12 +340,32 @@ const ruleForBabelLoader = (babelPlugins) => {
     return {
         id: 'babel-loader',
         test: /(\.js(x?)|\.ts(x?))$/,
-        // NOTE: Because our extensions are just folders containing source code, we need to ensure that the babel-loader processes them.
-        // This regex exclude everything in node_modules, but node_modules/extensions-*/ folders
-        exclude: new RegExp(
-            `node_modules\\${path.sep}(?!(@?[^\\${path.sep}]+\\${path.sep})?extension-).*`,
-            'i'
-        ),
+        // Determine if a module is an extension
+        exclude: (modulePath) => {
+            // Skip node_modules
+            if (!modulePath.includes('node_modules')) {
+                return false
+            }
+
+            // Extract package path from node_modules
+            const nodeModulesPath = modulePath.split('node_modules' + path.sep)[1]
+            if (!nodeModulesPath) {
+                return true
+            }
+
+            // Get the package name portion
+            const packageName = nodeModulesPath.split(path.sep)[0]
```

Comment:
Question: Are we actually expecting that package name would ever be undefined?

Reply:
I refactored the logic to use a normalized path for comparison